File: src/main/java/intro/_03_data_classes/Size.java
package intro._03_data_classes;

import java.util.Objects;

public class Size {

    private final int columns;
    private final int rows;

    public Size(int columns, int rows) {
        this.columns = columns;
        this.rows = rows;
    }

    public int getColumns() {
        return columns;
    }

    public int getRows() {
        return rows;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Size size = (Size) o;
        return getColumns() == size.getColumns() &&
                getRows() == size.getRows();
    }

    @Override
    public int hashCode() {
        return Objects.hash(getColumns(), getRows());
    }

    @Override
    public String toString() {
        return "Size{" +
                "columns=" + columns +
                ", rows=" + rows +
                '}';
    }
}

File: src/main/kotlin/workshop/kotlin/_03_data_classes/Size.kt
package workshop.kotlin._03_data_classes

import java.util.*

/**
 * You can check this class in its original version at:
 * https://github.com/Hexworks/zircon/blob/master/src/main/kotlin/org/codetome/zircon/api/Size.kt
 */
// TODO: create a `data class` `Size` based on the java Size class
class Size(val columns: Int, val rows: Int) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val size = o as Size?
        return columns == size!!.columns && rows == size.rows
    }

    override fun hashCode(): Int {
        return Objects.hash(columns, rows)
    }

    override fun toString(): String {
        return "Size{" +
                "columns=" + columns +
                ", rows=" + rows +
                '}'
    }
}